staging/src/kubevirt.io/api/instancetype/v1alpha2/types.go

```diff
@@ -77,11 +77,16 @@ type VirtualMachineClusterInstancetypeList struct {
 type VirtualMachineInstancetypeSpec struct {
 
 	// Required CPU related attributes of the instancetype.
+	// +optional
 	CPU CPUInstancetype `json:"cpu"`
 
 	// Required Memory related attributes of the instancetype.
+	// +optional
 	Memory MemoryInstancetype `json:"memory"`
 
+	// Resources describes the Compute Resources required by this vmi
+	Resources v1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
At an API schema level I think we need to make `CPU`, `Memory` and `Resources` optional and enforce the use of at least one in a validation webhook during submission. Making `CPU` and `Memory` optional while allowing an empty set of `Resources` is a little confusing IMHO, note `omitempty` is useless here as the field is required.